libevmasm/RuleList.h

```diff
@@ -57,6 +57,15 @@ template <class S> S shlWorkaround(S const& _x, unsigned _amount)
 	return u256((bigint(_x) << _amount) & u256(-1));
 }
 
+/// @returns k if _x == 2**k, nullopt otherwise
+inline std::optional<size_t> binaryLogarithm(u256 const& _x)
+{
+	if (!_x)
```

Comment:
```suggestion
	if (_x <= 1)
```
Probably better to avoid converting `mul(x, 1)` to `shl(0, x)`.

Reply:
The "mul(x, 1)' rule has higher priority, so there is no danger.

Reply:
Changed it to `_x == 0`, though.